container.go

```diff
@@ -257,38 +295,15 @@ func shortenPath(path string) string {
 }
 
 // untarFiles unpack only specified files from an archive to the destination directory.
-func untarFiles(src, dest string, files ...string) error {
+func untarFiles(src, dest string, files []string) error {
 	archiveFile, err := os.Open(src)
 	if err != nil {
 		return err
 	}
 	defer archiveFile.Close()
 
 	options := archive.TarOptions{
-		ExcludePatterns: []string{
-			"artifacts",
-			"ctr.log",
-			metadata.RootFsDiffTar,
-			metadata.NetworkStatusFile,
-			metadata.DeletedFilesFile,
-			metadata.CheckpointDirectory,
-			metadata.CheckpointVolumesDirectory,
-			metadata.ConfigDumpFile,
-			metadata.SpecDumpFile,
-			metadata.DevShmCheckpointTar,
-			metadata.DumpLogFile,
-			metadata.RestoreLogFile,
-		},
-	}
-
-	// Remove files from ExcludePatterns if they are present
-	for _, file := range files {
-		for i, pattern := range options.ExcludePatterns {
-			if pattern == file {
-				options.ExcludePatterns = append(options.ExcludePatterns[:i], options.ExcludePatterns[i+1:]...)
-				break
-			}
-		}
+		IncludeFiles: files,
```

Comment:
I understand what you are trying to do, but I am not sure it actually works as expected. Did you try it out?

When I last used this library, `IncludeFiles` did not work as I expected it during unpacking.

Reply:
Last I checked, it worked as expected. Let me double check once more.

Reply:
Thanks for pointing this out. I realised now that during tests, we move only the required files into the checkpoint folder before creating the archive. Since I used that to test this functionality, it appeared as though only the necessary files are being unpacked. I'll fix this.